src/main/java/net/javadiscord/javabot/systems/user_commands/MoveConversationCommand.java

```diff
@@ -0,0 +1,107 @@
+import net.dv8tion.jda.api.entities.GuildMessageChannel;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.commands.build.Commands;
+import net.dv8tion.jda.api.interactions.commands.build.OptionData;
+import net.dv8tion.jda.api.requests.restaction.MessageAction;
+import net.javadiscord.javabot.Bot;
+import net.javadiscord.javabot.data.config.guild.ModerationConfig;
+import net.javadiscord.javabot.util.Responses;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collections;
+
+/**
+ * <h3>This class represents the /move-conversation command.</h3>
+ */
+public class MoveConversationCommand extends SlashCommand {
+	private static final String MOVE_TO_MESSAGE = "``` ```\n\uD83D\uDCE4 %s has suggested to move **this conversation** over to %s\n%s\n\n``` ```";
+
+	private static final String MOVED_FROM_MESSAGE = "\uD83D\uDCE5 Conversation moved **here** from %s by %s\n%s";
+
+	/**
+	 * The constructor of this class, which sets the corresponding {@link net.dv8tion.jda.api.interactions.commands.build.SlashCommandData}.
+	 */
+	public MoveConversationCommand() {
+		setSlashCommandData(Commands.slash("move-conversation", "Suggest to move the current conversation into another channel!")
+				.addOptions(
+						new OptionData(OptionType.CHANNEL, "channel", "Where should the current conversation be continued?", true)
+								.setChannelTypes(ChannelType.TEXT, ChannelType.GUILD_NEWS_THREAD, ChannelType.GUILD_PRIVATE_THREAD, ChannelType.GUILD_PUBLIC_THREAD, ChannelType.VOICE, ChannelType.STAGE)
+				).setGuildOnly(true)
+		);
+	}
+
+	@Override
+	public void execute(@NotNull SlashCommandInteractionEvent event) {
+		OptionMapping channelMapping = event.getOption("channel");
+		if (channelMapping == null) {
+			Responses.replyMissingArguments(event).queue();
+			return;
+		}
+		if (event.getGuild() == null || event.getMember() == null) {
+			Responses.replyGuildOnly(event).queue();
+			return;
+		}
+		GuildMessageChannel channel = channelMapping.getAsChannel().asGuildMessageChannel();
+		if (event.getChannel().getIdLong() == channel.getIdLong()) {
+			Responses.warning(event, "Invalid Channel", "You cannot move the conversation to the same channel!").queue();
+			return;
+		}
+		if (isInvalidChannel(event.getMember(), channel)) {
+			Responses.warning(event, "Invalid Channel", "You're not allowed to move the conversation to %s", channel.getAsMention()).queue();
+			return;
+		}
+		event.deferReply(true).queue();
+		sendMovedFromChannelMessage(event, channel).queue(movedTo ->
+				sendMoveToChannelMessage(event, channel, movedTo).queue(movedFrom ->
+						editMovedFromChannelMessage(event, movedFrom, movedTo).queue(s ->
```

Comment:
Qe should probably also add a case for when sending the messages fails.

Reply:
Done